File: Benchmarking/src/gen_12_19.cpp
  #include <vector>
  #include <iostream>
  #include <chrono>
  struct s0 {
    std::vector<std::vector<double>> p0;
    std::vector<std::vector<double>> p1;
    s0(std::vector<std::vector<double>> v0, std::vector<std::vector<double>> v1): p0(v0), p1(v1) { }
  };
  struct s1 {
    s0 p0;
    s0 p1;
    s1(s0 v0, s0 v1): p0(v0), p1(v1) { }
  };
  struct s2 {
    std::vector<s0> p0;
    std::vector<std::vector<s1>> p1;
    s2(std::vector<s0> v0, std::vector<std::vector<s1>> v1): p0(v0), p1(v1) { }
  };
  double f76(const double &v0) {
    double v2 = v0;
    double v4 = v0;
    double v8 = v4;
    const double v1 = v8 / v2;
    v2 = v4;
    double v5 = v2;
    double v10 = v1;
    double v6 = v2;
    double v7 = v1;
    v5 = v8;
    double v20 = v5;
    const std::vector<double> v21 { v10, v6, v5, v7, v4 };
    std::vector<double> v32 = v21;
    v32[3] = v20;
    const double v29 = v32[0];
    return v29;
  }
  double f34(const double &v0) {
    const double v5 = f76(v0);
    const double v2 = f76(v5);
    double v1 = v5;
    const double v19 = v2 * v1;
    const double v16 = f76(v5);
    const double v29 = f76(v16);
    double v20 = v29;
    const std::vector<double> v13 { v20, v29, v19 };
    const double v23 = v13[0];
    return v23;
  }
  __attribute__((noinline))
  double f0(const std::vector<s0> &v0, const s2 &v1, const double &v2) {
    double v3 = v2;
    double v57 = v3;
    const double v19 = f34(v57);
    const double v41 = f76(v19);
    return v41;
  }
  int main() {
    std::vector<s0> v0({ { { { 0.0 }, { 1.0 } }, { { 2.0 }, { 3.0 }, { 4.0 } } }, { { { 5.0 }, { 6.0 } }, { { 7.0 }, { 8.0 }, { 9.0 } } }, { { { 10.0 }, { 11.0 } }, { { 12.0 }, { 13.0 }, { 14.0 } } } });
    s2 v1({ { { { { 15.0 }, { 16.0 } }, { { 17.0 }, { 18.0 }, { 19.0 } } } }, { { { { { { 20.0 }, { 21.0 } }, { { 22.0 }, { 23.0 }, { 24.0 } } }, { { { 25.0 }, { 26.0 } }, { { 27.0 }, { 28.0 }, { 29.0 } } } } } } });
    double v2(30.0);
    auto start = std::chrono::high_resolution_clock::now();
    double result;
    for (int i = 0; i < 1000; i ++) {
      result = f0(v0, v1, v2);
    }
    auto end = std::chrono::high_resolution_clock::now();
    double value = *((double*) &result);
    std::cout << value << "\n";
    std::cout << std::chrono::duration_cast<std::chrono::nanoseconds>(end-start).count();
    std::cout << "\n";
    return 0;
  }

File: Benchmarking/src/gen_12_19.swift
  import Dispatch
  struct s0 {
    var p0: [[Double]]
    var p1: [[Double]]
  }
  struct s1 {
    var p0: s0
    var p1: s0
  }
  struct s2 {
    var p0: [s0]
    var p1: [[s1]]
  }
  func f76(_ v0: Double) -> Double {
    var v2: Double = v0
    var v4: Double = v0
    var v8: Double = v4
    let v1: Double = v8 / v2
    v2 = v4
    var v5: Double = v2
    var v10: Double = v1
    var v6: Double = v2
    var v7: Double = v1
    v5 = v8
    var v20: Double = v5
    let v21: [Double] = [v10, v6, v5, v7, v4]
    var v32: [Double] = v21
    v32[3] = v20
    let v29: Double = v32[0]
    return v29
  }
  func f34(_ v0: Double) -> Double {
    let v5: Double = f76(v0)
    let v2: Double = f76(v5)
    var v1: Double = v5
    let v19: Double = v2 * v1
    let v16: Double = f76(v5)
    let v29: Double = f76(v16)
    var v20: Double = v29
    let v13: [Double] = [v20, v29, v19]
    let v23: Double = v13[0]
    return v23
  }
  @inline(never)
  func f0(_ v0: [s0], _ v1: s2, _ v2: Double) -> Double {
    var v3: Double = v2
    var v57: Double = v3
    let v19: Double = f34(v57)
    let v41: Double = f76(v19)
    return v41
  }
  func benchmark() {
    let v0: [s0] = [s0(p0: [[0.0], [1.0]], p1: [[2.0], [3.0], [4.0]]), s0(p0: [[5.0], [6.0]], p1: [[7.0], [8.0], [9.0]]), s0(p0: [[10.0], [11.0]], p1: [[12.0], [13.0], [14.0]])]
    let v1: s2 = s2(p0: [s0(p0: [[15.0], [16.0]], p1: [[17.0], [18.0], [19.0]])], p1: [[s1(p0: s0(p0: [[20.0], [21.0]], p1: [[22.0], [23.0], [24.0]]), p1: s0(p0: [[25.0], [26.0]], p1: [[27.0], [28.0], [29.0]]))]])
    let v2: Double = 30.0
    let start = DispatchTime.now().uptimeNanoseconds
    var result: Double = 30.0
    for _ in 1...1000 {
      result = f0(v0, v1, v2)
    }
    let end = DispatchTime.now().uptimeNanoseconds
    print(result)
    print(end - start)
  }
  benchmark()
